Autoencoder/Autoencoder.swift

```
@@ -0,0 +1,285 @@
+    let labelData = try Data(contentsOf: labelFolderURL).dropFirst(8)
+    let images = imageData.map { Float($0) }
+    let labels = labelData.map { Int32($0) }
+    let rowCount = Int32(labels.count)
+    let columnCount = Int32(images.count) / rowCount
+    
+    print("Constructing the data tensors.")
+    let imagesTensor = Tensor(shape: [rowCount, columnCount], scalars: images)
+    let labelsTensor = Tensor(labels)
+    return (imagesTensor, labelsTensor)
+}
+
+func plot(tensor: [Float], labels: Tensor<Int32>, step: Int) {
+    // Import Python modules
+    let matplotlib = Python.import("matplotlib")
+    // Turn off using display on server / linux
+    matplotlib.use("Agg")
+    
+    let np = Python.import("numpy")
+    let plt = Python.import("matplotlib.pyplot")
+    
+    // Create figure
+    let ax = plt.gca()
+    // Create colors
+    let linspace = np.linspace(start: 0, stop: 1, num: 10, endpoint: true, retstep: false)
+    let colorMap = plt.get_cmap("RdBu")
+    let colors = np.r_[linspace, linspace]
+    let mappedColors = colorMap(colors)
+    for element in 0..<10 {
+        let xElementIndex = labels.scalars.enumerated().filter { $0.element == element}.map { 2 * $0.offset + 0}
+        let yElementIndex = labels.scalars.enumerated().filter { $0.element == element}.map { 2 * $0.offset + 1}
+        let x = tensor.enumerated().filter { xElementIndex.contains($0.offset) }.map { $0.element }
+        let y = tensor.enumerated().filter { yElementIndex.contains($0.offset) }.map { $0.element }
+        ax.scatter(x: x, y: y, color: mappedColors[element], label: "\(element)", s: 20, alpha: 0.8)
+    }
+    ax.set_xlabel("Autoencoder clasterization.")
+    ax.legend()
+    //plt.axis.dynamicallyCall(withArguments:[-0.3, 0.3, -0.3, 0.3])
+    if !FileManager.default.fileExists(atPath: outputFolder) {
+        try? FileManager.default.createDirectory(atPath: outputFolder, withIntermediateDirectories: false, attributes: nil)
```

This `try?` is gonna return an unused value. And directory creation failure is not handled here.

Handling it is not super important for today, so you can just use `try!`.